src/eon_type.erl

```diff
@@ -286,7 +286,7 @@ typecheck(#spec{term=Term0, type=Type, p_have=P_have}) ->
         true ->
           case Type:extra_validation(Term, P_have) of
             ok -> Term;
-            {error, Reason} -> {error, {untypable, Reason}}
+            {error, Reason} -> {error, {untypable, [{x, Type, x, Reason, []}]}}
```

Comment:
Maybe you can add a comment about this hack where you return the `Reason` as the `Term`? I can see how it makes the root cause easier to understand but this part of the code looks off since it differs from this `{Key, Type, Type:name(), Term, Rsn}` that is usually returned.